src/libs/core/CMakeLists.txt

```diff
@@ -27,14 +27,18 @@ else()
     target_link_libraries(Core Qt5::WebEngine)
 endif()
 
-find_package(LibArchive REQUIRED)
-
 # Set LibArchive_INCLUDE_DIR on macOS when LibArchive is installed with Homebrew.
 # See https://github.com/Homebrew/legacy-homebrew/issues/21415.
 if(APPLE AND NOT LibArchive_INCLUDE_DIR)
-  set(LibArchive_INCLUDE_DIR "/usr/local/opt/libarchive/include")
+  if(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
+    set(LibArchive_INCLUDE_DIR "/usr/local/opt/libarchive/include")
+  elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64")
+    set(LibArchive_INCLUDE_DIR "/opt/homebrew/opt/libarchive/include")
```

Comment:
Is there a way to detect Homebrew installation rather than hard-coding the path? Also what if `arm64` has LibArchive installed not from Homebrew? I'd expect `find_package` to take care of this?

Reply:
Yes, using `find_package` looks more elegant. Let me do some trial for different LibArchive installation cases:
- Homebrew installed, both x86_64 and arm64
- Installed to system path, but not using homebrew
- User compiled and installed, but not installed to system path

Reply:
> Is there a way to detect Homebrew installation rather than hard-coding the path? Also what if `arm64` has LibArchive installed not from Homebrew? I'd expect `find_package()` to take care of this?

Actually there is a `libarchive.tdb` library file in subdirectories of macOS SDK. Previously I set `LibArchive_INCLUDE_DIR` to homebrew's one, then call `find_package()`, but the result is mismatched pair: homebrew installed headers, macOS SDK provided library file.

The new force-pushed commit now find the homebrew installed headers + library file.

Reply:
The installation of LibArchive, does not provide a `LibArchiveConfig.cmake` or `libarchive-config.cmake` file. It provides `libarchive.pc` file, which, can be used to find LibArchive headers and library files by invoking wrapper of `pkg-config`, but looks like not very simple.